### PR TITLE
Cheeseman's HW3

### PR DESCRIPTION
**Pull request by cheeseman1997**
This is my attempt at homework 3. Hopefully, it will be good...

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4853259&bid=16101341)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using static System.Math;
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
+        {
+            string cont = "y";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'cont'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4853259)

File: Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public class Lesson3Tests
         [InlineData(100, 10, 1)]
         public void CalculateBmi_With_ValidInput_Returns_Expected(float weightKg, float heightM, float expectedBmi)
```
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs]: PLease amend all the changes made to the test project. You're not supposed to touch it.

**Comment by ignasposka** [comment]
Your build is failing because pipeline cannot access class `Lesson3`. Make it public.
`public class Lesson3`

File: Src/BootCamp.Chapter/Checks.cs [hunk 3]
```diff
@@ -4,38 +4,35 @@
...
-            return 0;
+
+            return Lesson3.PromptInteger(message);
         }
 
         public static string PromptString(string message)
         {
-            // To do: call your implementation. 
-            return "";
+
```
Comment (ignasposka) [inline Src/BootCamp.Chapter/Checks.cs]: You don't need to have so many empty lines, remove them.

File: Src/BootCamp.Chapter/Checks.cs [hunk 4]
```diff
@@ -4,38 +4,35 @@
...
 {
-    /// <summary>
-    /// Test class is used to test your implementation.
-    /// Each homework will have a set of steps that you will have to do.
-    /// You can name your functions however you want, but to validate your solution, place them here.
-    /// DO NOT CALL FUNCTIONS FROM TESTS CLASS
-    /// DO NOT IMPLEMENT FUNCTIONS IN TESTS CLASS
-    /// TESTS CLASS FUNCTIONS SHOULD ALL HAVE 1 LINE OF CODE
-    /// </summary>
+    namespace BootCamp.Chapter
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Checks.cs]: This doesn't seem like it was intended.

File: Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs [hunk 5]
```diff
@@ -56,7 +56,7 @@ public void PromptFloat_PrintsMessage_And_ReturnsFloat(string input, float expec
         {
             var consoleOutput = ConsoleStub.StubConsole(input);
 
-            var convertedInput = Checks.PromptFloat(PromptMessage);
+            var convertedInput = Lesson3.PromptFloat(PromptMessage);
```
Comment (ellmaster) [inline Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs]: You probably changed this by auto-refactoring something. Be careful when doing so and let Visual Studio show you which files are going to be changed in which way!

File: Src/BootCamp.Chapter/Lesson3.cs (hunk 1 above)
Comment (ellmaster) [reply Src/BootCamp.Chapter/Lesson3.cs]: This is a minor issue. Still line 12 will override the initial value and makes it unneccessary in the first place.

**Comment by Almantask** [comment]
Expired

**Pull request closed without merging**